src/pushNotifications/widget/pushNotifications.js

```diff
@@ -297,17 +297,41 @@ define([
                 deviceRegistration.set("DeviceType", "iOS");
             }
 
-            // We commit the DeviceRegistration object to trigger the backend process.
-            // This is the only time we can commit the object, because it will be deleted in the AfterCommit event.
-            mx.data.commit({
-                mxobj: deviceRegistration,
-                callback: dojoLang.hitch(this, function() {
-                    logger.debug("Registered device with ID " + deviceRegistration.get(this.REGISTRATION_ID_ATTRIBUTE));
-                }),
-                error: function(e) {
-                    logger.error("Error occurred attempting to register device: " + e);
-                }
-            });
+            // CC Jan 10, 2018 - only commit the DeviceRegistration if the registration id has changed or does not yet exist
+            var lastRegistrationId = localStorage.getItem("mx-registration-id");
+            if (!lastRegistrationId || lastRegistrationId !== deviceRegistration.get(this.REGISTRATION_ID_ATTRIBUTE)) {
```

Comment:
We have 'this._registrationId'. Why not use that?

Reply:
because `this._registrationId` is always the most recent registration, and in order to prevent the widget from syncing each time a new page is shown that has this widget, it should only sync when the registration id changes. That's why it stored the id in localstorage when it successfully registers, and then compares the new registration id to that old value before synchronizing again (since syncing can take a while)